packages/extension-dapp/src/bundle.ts

```diff
@@ -60,10 +63,22 @@ function filterAccounts (list: InjectedAccount[], genesisHash?: string | null, t
 
 /** @internal retrieves all the extensions available on the window */
 function getWindowExtensions (originName: string): Promise<InjectedExtension[]> {
+  let extensions: Record<string, InjectedWindowProvider>;
+  
+  /** If originName indicates enabling only snap */
+  if (['onlysnap', 'only_snap', 'snaponly', 'snap_only'].includes(originName.toLowerCase())) {
```

Comment:
this looks hacky, why would one only want the snap. Devs can pick the snaps only should they wish to. I'd remove this part.

Reply:
I have add inline comment as follows:

/** Since web3Enable enables all available extensions, which is the default behavior
   *  for Polkadot JS apps, some dapps, like the Staking dashboard, provide an extension
   *  list where users can choose specific extensions to enable. Therefore, we utilize
   *  "Snap only" to inject snaps as an additional feature, suitable for such dapps.
   * */